### ISSUE TITLE
Does this work with generic non-SVG components?

### ISSUE THREAD

**Issue by AdityaAnand1** [uber/react-digraph#262]
I would like to use my own react components as graph nodes instead of SVGs. Is this supported?

**Comment by ajbogh** [comment]
You can pass a renderNode function that returns a react component. These components must be SVGs, but you can use the foreignObject element to wrap HTML data as well. A few people have used this method for their projects.

https://developer.mozilla.org/en-US/docs/Web/SVG/Element/foreignObject

**Comment by tony-garcia** [comment]
Yes I've used foreignObject to render HTML nodes using React and it works quite well.
